var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -65,6 +65,8 @@ class Trilinos(CMakePackage):
     version('11.14.2', 'e7c3cdbbfe3279a8a68838b873ad6d51')
     version('11.14.1', 'b7760b142eef66c79ed13de7c9560f81')
 
+    variant('allpkgs', default=True,
```

Comment:
i would not add this. The reasons is that you duplicate options for anyone who use trilinos, i.e. if a package needs `epetra`, then it needs either `trilinos+epetra` or `trilinos+allpkgs`. IMO that's hard to manage and over-complicate things. Just remove this and use `default=True` in all extra variants below.